packages/block-editor/src/hooks/block-binding-support/index.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * WordPress dependencies
+ */
+import { addFilter } from '@wordpress/hooks';
+
+/**
+ * Internal dependencies
+ */
+import withBlockBindingSupport from './with-block-binding-support';
+
+export const BLOCK_BINDINGS_ALLOWED_BLOCKS = {
+	'core/paragraph': [ 'content' ],
+	'core/heading': [ 'content' ],
+	'core/image': [ 'url', 'title', 'alt' ],
+	'core/button': [ 'url', 'text', 'linkTarget' ],
+};
+
+export function isItPossibleToBindBlock( blockName ) {
+	return blockName in BLOCK_BINDINGS_ALLOWED_BLOCKS;
+}
```

Comment:
Nit: Should we call this `hasPossibleBlockBinding` or something like that?

Reply:
The function returns true when binding the block attributes with the external source is possible. It's based on the block `name,` and the current criteria relies only on a hardcoded blocks list. It doesn't inspect the block attributes to check if the block "has possible binding".
Having said that, I am more than happy to rename it.

Reply:
It's a private method, so naming doesn't really matter at the moment. We can come up with final name if we decide to make it public :)

Reply:
Actually... thinking, well, I think your suggestion makes total sense. However, we'd probably like to have both checkings. One at the registering block level and another one in the block visualization context (edit function).

Reply:
https://github.com/WordPress/gutenberg/pull/58895/commits/56dc165707ebc832967e9026b090a2690d4399f9